src/strategies/staked-defi-balance/index.ts

```diff
@@ -15,40 +18,80 @@ export async function strategy(
-  const abi = options.methodABI;
-
-  const stakingCalls = addresses.map((address) => {
-    return [
-      stakingPoolContractAddress,
-      'stakeOf',
-      [options.tokenContractAddress, address]
-    ];
-  });
+  const maxContractsPerStrategy = 5;
+  if (options.length > maxContractsPerStrategy) {
+    throw new Error(
+      'Maximum of 5 contracts allowed per strategy, see details: https://github.com/snapshot-labs/snapshot-strategies#code'
+    );
+  }
+  const addressScores = {};
 
-  const stakes = await multicall(network, provider, abi, stakingCalls, {
-    blockTag
-  });
+  for (const params of options) {
+    const paramNetwork = network.toString();
+    const paramSnapshot = typeof snapshot === 'number' ? snapshot : 'latest';
 
-  const stakesMapped = {};
-  for (let i = 0; i < addresses.length; i++) {
-    stakesMapped[getAddress(addresses[i])] = stakes[i][0];
-  }
+    const stakingPoolContractAddress = params.stakingPoolContractAddress;
+    let abi: ABI;
+    switch (params.stakingType) {
+      case 'open':
+        abi = openStakingAbi[0] as ABI;
+        break;
+      case 'standard':
+        abi = standardStakingAbi[0] as ABI;
+        break;
+      default:
+        throw new Error(`Invalid stakingType: ${params.stakingType}`);
+    }
+
+    const stakingCalls = addresses.map((address) => {
+      const inputs = abi.inputs.map((input) => {
+        if (input.name === 'id') {
+          return params.tokenContractAddress;
+        } else if (input.name === 'staker' || input.name === 'account') {
+          return address;
+        }
+      });
+      return [stakingPoolContractAddress, abi.name, inputs];
+    });
+
+    const stakes = await multicall(
+      paramNetwork,
+      getProvider(paramNetwork),
+      [abi],
+      stakingCalls,
+      { blockTag: paramSnapshot }
+    );
 
-  const addressScores = Object.fromEntries(
-    addresses.map((address) => {
+    const stakesMapped = {};
+    for (let i = 0; i < addresses.length; i++) {
+      stakesMapped[getAddress(addresses[i])] = stakes[i][0];
+    }
+
+    addresses.forEach((address) => {
       const normalizedAddress = getAddress(address);
       const stakedBalance = stakesMapped[normalizedAddress];
       const formattedStakedBalance = parseFloat(
-        formatUnits(stakedBalance, options.decimals)
+        formatUnits(stakedBalance, params.decimals)
       );
-      return [
-        normalizedAddress,
-        stakedBalance.gte(options.minStakedBalance) ? formattedStakedBalance : 0
-      ];
-    })
+
+      // Initialize address score if it doesn't exist
+      if (!addressScores[normalizedAddress]) {
+        addressScores[normalizedAddress] = 0;
+      }
+
+      addressScores[normalizedAddress] += formattedStakedBalance;
+    });
+  }
+
+  // Filter out addresses that have a total staked balance less than the minStakedBalance
+  const minStakedBalance = parseFloat(
+    formatUnits(options[0].minStakedBalance, options[0].decimals)
```

Comment:
here it checks only first params minStakedBalance @taha-abbasi

Reply:
The balance is the same in each contract based on how this strategy will be used so no need to check each one. @ChaituVR